ros/src/tl_detector/light_classification/tl_classifier.py

```diff
@@ -41,3 +47,41 @@ def get_colored_area(self, image, lower, upper):
 
         return area, extracted_image
 
+
+class TLClassifier(object):
+
+	def __init__(self):
+
+		with open('light_classification/saved_models/model128.json', 'r') as f:
+			loaded_model_json = f.read()
+		self.model = model_from_json(loaded_model_json)
+		self.model.load_weights('light_classification/saved_models/weights128.hdf5')
+		self.graph = tf.get_default_graph()
+
+	def get_classification(self, image):
+		"""
+		Determines the color of the traffic light in image
+
+		Args:
+			image (cv::Mat): image containing the traffic light
+
+		Returns:
+			int: ID of traffic light color (specified in styx_msgs/TrafficLight)
+		"""
+		processed_image = self.process_image(image)
+		with self.graph.as_default():
+			predicted_class_id = self.model.predict_classes(processed_image, batch_size = 1, verbose = 0)
```

Comment:
Where did you find info about method model.predict_classes()? I use Keras daily and have never seen it before. Had to look up the source to find it :D

Reply:
I found it here( https://keras.io/ja/models/sequential/#the-sequential-model-api)
Only Japanese page has the documentation on it...